tests/playwright-test/esm.spec.ts

```diff
@@ -775,3 +776,44 @@ test('should exit after merge-reports', async ({ runInlineTest, mergeReports })
   const { exitCode } = await mergeReports(test.info().outputPath('blob-report'), undefined, { additionalArgs: ['-c', 'merge.config.ts'] });
   expect(exitCode).toBe(0);
 });
+
+test('should not trip over directory imports resolved to paths', async ({ runInlineTest, writeFiles }, testInfo) => {
+  test.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/31811' });
+
+  await writeFiles({
+    '@acme/lib/package.json': JSON.stringify({
+      'name': '@acme/lib',
+      'type': 'module',
+      'exports': {
+        '.': './lib.js',
+      },
+    }),
+    '@acme/lib/lib.js': `
+      export const greet = () => console.log('hello playwright');
+    `,
+    'tests/hello.test.ts': `
+      import { greet } from '@acme/lib';
+      import { test } from '@playwright/test';
+      test('hello', async ({}) => {
+        greet();
+      });
+    `,
+    'tests/tsconfig.json': JSON.stringify({
+      compilerOptions: {
+        'paths': {
+          '@acme/*': ['../@acme/*'],
+        }
+      }
+    })
+  });
+
+  await fs.mkdir(testInfo.outputPath('node_modules', '@acme'), { recursive: true });
+  await fs.symlink('../../@acme/lib', testInfo.outputPath('node_modules', '@acme', 'lib'));
```

Comment:
Because `compilerOptions#paths` is required to trigger the faulty behavior, and that's typically used in a monorepo kind of scenario. At least that's what happened in the user repro. Also, from my experience working on bundlers + their Node.js resolution, you can never have enough tests with symlinks.